src/contentscript/overlay/root/components/Profile/HeaderLogIn/index.tsx

```diff
@@ -115,9 +115,9 @@ export const HeaderLogIn: FC<HeaderLogInProps> = (props: HeaderLogInProps) => {
           setWalletAccount(newDescriptors.account && truncateEthAddress(newDescriptors.account))
         }
         if (newDescriptors.type !== 'dapplets') {
-          setWalletIcon(newDescriptors.meta.icon)
-        } else {
           setWalletIcon(walletIcons[newDescriptors.type])
+        } else {
+          setWalletIcon(null)
```

Comment:
As I understood you have to check this condition because the built-in "dapplets" wallet doesn't have an icon. You can declare the icon here: https://github.com/dapplets/dapplet-extension/blob/203843f93a8093632a524efad3ec164b10d56018/src/background/wallets/ethereum/dapplets.ts#L99
As here: https://github.com/dapplets/dapplet-extension/blob/203843f93a8093632a524efad3ec164b10d56018/src/background/wallets/near/near/index.ts#L110
And remove the condition.